pkg/controller/studyjob/utils.go

```diff
@@ -206,6 +207,91 @@ func contains(l []string, s string) bool {
 	return false
 }
 
+func validateHPJob(instance *katibv1alpha1.StudyJob) error {
+	log.Printf("Validation for the HP job")
+	if instance.Spec.ParameterConfigs == nil {
+		return fmt.Errorf("No Spec.ParameterConfigs specified")
+	}
+	err := validateParameterConfigs(instance.Spec.ParameterConfigs, jobTypeHP, instance.Spec.SuggestionSpec.SuggestionAlgorithm)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateNASJob(instance *katibv1alpha1.StudyJob) error {
+	log.Printf("Validation for the NAS job")
+	if instance.Spec.NasConfig == nil {
+		return fmt.Errorf("No Spec.NasConfig specified")
+	}
+
+	if reflect.DeepEqual(instance.Spec.NasConfig.GraphConfig, katibv1alpha1.GraphConfig{}) {
+		return fmt.Errorf("Missing GraphConfig in NasConfig: %v", instance.Spec.NasConfig)
+	}
+
+	if instance.Spec.NasConfig.GraphConfig.InputSize == nil {
+		return fmt.Errorf("Missing InputSize in NasConfig.GraphConfig: %v", instance.Spec.NasConfig.GraphConfig)
+	}
+
+	if instance.Spec.NasConfig.GraphConfig.OutputSize == nil {
+		return fmt.Errorf("Missing OutputSize in NasConfig.GraphConfig: %v", instance.Spec.NasConfig.GraphConfig)
+	}
+
+	if instance.Spec.NasConfig.GraphConfig.NumLayers == 0 && instance.Spec.SuggestionSpec.SuggestionAlgorithm == suggestionAlgorithmRL {
+		return fmt.Errorf("Missing NumLayers in NasConfig.GraphConfig: %v", instance.Spec.NasConfig.GraphConfig)
+	}
+
+	if instance.Spec.NasConfig.Operations == nil {
+		return fmt.Errorf("Missing Operations in NasConfig: %v", instance.Spec.NasConfig)
+	}
+
+	for _, op := range instance.Spec.NasConfig.Operations {
+		if op.OperationType == "" {
+			return fmt.Errorf("Missing OperationType in Operation: %v", op)
+		}
+		if op.ParameterConfigs == nil {
+			return fmt.Errorf("Missing ParameterConfig in Operation: %v", op)
+		}
+		err := validateParameterConfigs(op.ParameterConfigs, jobTypeNAS, instance.Spec.SuggestionSpec.SuggestionAlgorithm)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateParameterConfigs(parameterConfigs []katibv1alpha1.ParameterConfig, jobType string, suggestionAlgorithm string) error {
+	for _, pc := range parameterConfigs {
+		if pc.Name == "" {
+			return fmt.Errorf("Missing Name in ParameterConfig: %v", pc)
+		}
+		if pc.ParameterType == "" {
+			return fmt.Errorf("Missing ParameterType in ParameterConfig: %v", pc)
+		}
+		if reflect.DeepEqual(pc.Feasible, katibv1alpha1.FeasibleSpace{}) {
+			return fmt.Errorf("Missing Feasible in ParameterConfig: %v", pc)
+		}
+		if pc.ParameterType == katibv1alpha1.ParameterTypeCategorical {
```

Comment:
Not only categorical but discrete.

Reply:
In discrete we also use only List structure or something else?

Reply:
Currently discrete only need List. But we may need more type for discretes, int or double.
Please treated in the same way as Categorical for now.

Reply:
Done